engine/table/src/main/java/io/deephaven/engine/table/impl/by/PartitionByChunkedOperator.java

```diff
@@ -411,12 +477,19 @@ private boolean appendShifts(@NotNull final LongChunk<? extends RowKeys> preShif
         }
         if (builder == null) {
             final RowSet tableRowSet = tables.getUnsafe(destination).getRowSet();
-            final RowSet removedRowSet = removedRowSets.getUnsafe(destination);
+            final RowSetBuilderRandom removedRowSetBuilder = removedBuilders.getUnsafe(destination);
             final RowSet preShiftKeys;
-            if (removedRowSet == null) {
+            if (removedRowSetBuilder == null) {
                 preShiftKeys = tableRowSet.copy();
             } else {
-                preShiftKeys = tableRowSet.minus(removedRowSet);
+                // we want to keep the builder in `removedRowSetBuilders` ready for a call to `extractAndClearRowSet` so
```

Comment:
Rebuilding the `RowSet` this way is unfortunate, and we should avoid it. We could instead use a dummy builder that "holds" a single `RowSet` to return on `build()` and throws `UnsupportedOperationException` for all other methods.